# Домашнє завдання до теми "Основи роботи з рядками та математичні операції"

1. Створіть змінні `a`, `b`, `c` та `d`, задайте їм значення відповідно 101, 210, 111, 402, та обчисліть з допомогою Python наступний вираз:  $$ \frac{(a+b) * c}{d} $$

**Важливо**: тут ми маємо обчислити значення саме з використанням змінних, а не просто чисел, як би ми робили у калькуряторі. Змінні потрібні аби за бажання ми могли змінити їх значення та не переписувати код обчислення виразу, бо він буде працювати для різних значень змінних.

In [1]:
a = 101
b = 210
c = 111
d = 402

res = ((a + b) * c) / d
print(res)

85.8731343283582


Очікувана відповідь: 85.8731343283582

2. Якою буде остача від ділення числа 11283 на 89? Обчисліть з допомогою Python.

In [4]:
11283 % 89

69

3. Скільки цілих разів поміститься число 89 в 11283? (Або що те саме: який буде результат цілочисельного ділення 11283 на 89?)

In [6]:
11283 // 89

126

4. Запишіть у змінну `name` ваше імʼя.

In [8]:
name = 'Olena'

5. Запишіть у змінну `surname` ваше прізвище.

In [9]:
surname ='Bublivska'

6. Сконкатенуйте імʼя та прізвище в одну фразу через пробіл. Запишіть результат у нову змінну `name_surname` та виведіть результат на екран.  

In [10]:
name_surname = name + ' ' + surname
print(name_surname)

Olena Bublivska


Очікувана відповідь (тільки з вашими даними): 'Hanna Pylieva'

7. Зробіть усі літери змінної `name_surname` заглавними використовуючи метод рядків в Python:

In [11]:
name_surname_upper = name_surname.upper()
print(name_surname_upper)

OLENA BUBLIVSKA


Очікувана відповідь (тільки з вашими даними): 'HANNA PYLIEVA'

8. Виведіть третю літеру вашого `name_surname`. Тут уважно, треба саме третя літера. Згадайте, з якого значення починається індексування значень в Python.

In [15]:
name_surname[2]

'e'

Очікувана відповідь (тільки з вашими даними): 'n'

9. Виведіть першу та останню літеру вашого прізвища.

In [16]:
first_letter = name_surname[0]
last_letter = name_surname[-1]

print(first_letter)
print(last_letter)

O
a


Очікувана відповідь (тільки з вашими даними): ('P', 'a')

10. Виведіть довжину вашого прізвища.

In [17]:
len(surname)

9

Очікувана відповідь (тільки з вашими даними): 7


11. Виведіть кількість літер `a` у столиці Мадагаскару `Antananarivo`. Важливо врахувати всі літери, як великі, так і маленькі.

In [20]:
count_a = "Antananarivo".lower().count('a')

print(count_a)

4


Очікувана відповідь: 4

12. Створіть рядок `my_string` зі значенням: "The quick brown fox jumps over the lazy dog!!!".   
Перевірте, чи починається рядок `my_string` з "The" і чи закінчується він "dog!". Виведіть результати на екран.

In [21]:
my_string = 'The quick brown fox jumps over the lazy dog!!!'
starts_with_the = my_string.startswith('The')
ends_with_dog = my_string.endswith('dog!')

print("Рядок my_string починається з 'The':", starts_with_the)
print("Рядок my_string закінчується на 'dog!':", ends_with_dog)

Рядок my_string починається з 'The': True
Рядок my_string закінчується на 'dog!': False


13. Розділіть рядок `my_string` на список окремих слів, а потім з'єднайте всі слова через нижнє підкреслювання. Виведіть результат на екран.

In [22]:
my_string = "The quick brown fox jumps over the lazy dog!!!"

words_list = my_string.split()

joined_string = "_".join(words_list)

print(joined_string)

The_quick_brown_fox_jumps_over_the_lazy_dog!!!


Очікуваний результат: `The_quick_brown_fox_jumps_over_the_lazy_dog!!!`